src/tests/tracing/eventpipe/userevents/userevents.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Etlx;
+
+namespace Tracing.Tests.UserEvents
+{
+    public class UserEventsTest
+    {
+        private static readonly string trace = "trace.nettrace";
+        private const int SIGINT = 2;
+
+        [DllImport("libc", SetLastError = true)]
+        private static extern int kill(int pid, int sig);
+
+        public static int Main(string[] args)
+        {
+            if (args.Length > 0 && args[0] == "tracee")
+            {
+                UserEventsTracee.Run();
+                return 0;
+            }
+
+            return TestEntryPoint();
+        }
+
+        public static int TestEntryPoint()
+        {
+            string appBaseDir = AppContext.BaseDirectory;
+            string recordTracePath = Path.Combine(appBaseDir, "record-trace");
+            string scriptFilePath = Path.Combine(appBaseDir, "dotnet-common.script");
+            string traceFilePath = Path.Combine(appBaseDir, trace);
+
+            if (!File.Exists(recordTracePath) || !File.Exists(scriptFilePath))
+            {
+                Console.WriteLine("record-trace or dotnet-common.script not found. Test cannot run.");
```

Comment:
Its helpful to print the exact path we didn't find. (In general its helpful for the test logging to err towards being overly verbose to make failure diagnosis easier)